src/platforms/macos/daemon/macosdaemonserver.cpp

```diff
@@ -43,20 +49,62 @@ int MacOSDaemonServer::run(QStringList& tokens) {
 
   MacOSDaemon daemon;
 
-  DaemonLocalServer server(qApp);
-  if (!server.initialize()) {
+  QLocalServer server(qApp);
+  server.setSocketOptions(QLocalServer::WorldAccessOption);
+  connect(&server, &QLocalServer::newConnection, [&]() {
+    logger.debug() << "New connection received";
```

Comment:
Adding a bunch to this class. I'm likely missing it, but I don't see where this work was done before. Was this moved from another area of the code?

Reply:
This used to be performed by the `DaemonLocalServer` It was only used in two places: `MacOSDaemonServer` and `WindowsDaemonServer` and didn't find that the class did enough to warrant its own existence (it really was just a wrapper around `QLocalServer`

It also didn't work for the `MockDaemon` due to the lack of a hardcoded path, so I found it cleaner to just get rid of the class and migrate its implementation directly into the Mac and Windows daemons.